src/PartitionedArrays.jl

```diff
@@ -82,6 +82,18 @@ function PartitionedArrays.PVector(v::PETScVector,ids::PRange,::MPIBackend)
   PVector(values,ids)
 end
 
+function Base.copy!(a::PVector,petsc_vec::Vec)
+  aux=PETScVector()
+  aux.vec[] = petsc_vec.ptr
```

Comment:
When `aux` goes of of the scope, I am afraid that the gc will call `Finalize` on `aux` calling `VecDestroy` on `petsc_vec`

As a general policy,  I would remove all overloads of Julia methods involving low level petsc types like `Vec` ( I think we only have `copy!`, to confirm).  Low level petsc objects, just need to be usable via the low level C wrappers in the `PETSC` module.  Otherwise, expect all kinds of problems.

Reply:
> When aux goes of of the scope, I am afraid that the gc will call Finalize on aux calling VecDestroy on petsc_vec

I dont see why this is the case. If am I not missing anything, `aux` can be GCed (indeed it can be right after `copy!`) without affecting to `petsc_vec` at all. Note that `aux` just temporarily stores the memory address of `petsc_vec`. Am I saying anything wrong?

> As a general policy, I would remove all overloads of Julia methods involving low level petsc types like Vec ( I think we only have copy!, to confirm). Low level petsc objects, just need to be usable via the low level C wrappers in the PETSC module. Otherwise, expect all kinds of problems.

I think that at this point we should start discussing why I needed this. Two reasons:

1. The PETSc residual and jacobian call backs pass us `Mat` and `Vec` objects. 
2. There is currently no constructor of `PETScMatrix` nor `PETScVector` that allows you to create them out of `Mat` and `Vector` objects, resp.

I though how to address 2. However, the solutions that came into my mind just did not fit into the way `PETScMatrix` nor `PETScVector` data types are currently designed. These assume a composition relationship among them and the corresponding `Mat` and `Vec` objects (e.g., the finalizer  calls `VecDestroy`). I found too much to modify the design of  `PETScMatrix` nor `PETScVector`  to reflect this use case (and expose it to users) given that this need is quite targeted, only for implementing the nonlinear solver. Perhaps there are more clever solutions.

Reply:
> When aux goes of of the scope, I am afraid that the gc will call Finalize on aux calling VecDestroy on petsc_vec

> I dont see why this is the case. If am I not missing anything, aux can be GCed (indeed it can be right after copy!) > without affecting to petsc_vec at all. Note that aux just temporarily stores the memory address of petsc_vec. Am I saying anything wrong?

Ok, I think I now understand what you mean! Note that we are not actually installing the finalizer on this instance of `PETScVector`. Thus the garbage collector cannot call `VecDestroy`, right? Anyway ... I agree this is a bit dirty. This stems from the need mentioned in my previous comment.

Reply:
Solved in d2f566f77fd173d0ad939cca2b96c4ffd6de22ed